tests/workflows/basic_tool_calling_node/tests/test_workflow.py

```diff
@@ -0,0 +1,323 @@
+
+    uuid4_generator = mock_uuid4_generator("vellum.workflows.nodes.displayable.bases.inline_prompt_node.node.uuid4")
+    first_call_input_id = uuid4_generator()
+    first_call_input_id_2 = uuid4_generator()
+    second_call_input_id = uuid4_generator()
+    second_call_input_id_2 = uuid4_generator()
+
+    # GIVEN a get current weather workflow
+    workflow = GetCurrentWeatherWorkflow()
+
+    # WHEN the workflow is executed
+    terminal_event = workflow.run()
+
+    # THEN the workflow is executed successfully
+    assert terminal_event.name == "workflow.execution.fulfilled"
+    assert (
+        terminal_event.outputs.text
+        == "Based on the function call, the current temperature in Miami is 70 degrees metric."
+    )
+    assert terminal_event.outputs.chat_history == [
+        ChatMessage(text="Hello, how are you?", role="USER"),
+        ChatMessage(text="I'm good, thank you!", role="ASSISTANT"),
+        ChatMessage(
+            text=None,
+            role="FUNCTION",
+            content=FunctionCallChatMessageContent(
+                type="FUNCTION_CALL",
+                value=FunctionCallChatMessageContentValue(
+                    name="get_current_weather",
+                    arguments={"location": "Miami", "unit": "metric"},
+                    id="call_7115tNTmEACTsQRGwKpJipJK",
+                ),
+            ),
+        ),
+        ChatMessage(
+            text="The current weather in Miami is sunny with a temperature of 70 degrees metric.",
+            role="FUNCTION",
+        ),
+        ChatMessage(
+            text="Based on the function call, the current temperature in Miami is 70 degrees metric.",
+            role="ASSISTANT",
+        ),
+    ]
+
+    first_call = vellum_adhoc_prompt_client.adhoc_execute_prompt_stream.call_args_list[0]
+    assert first_call.kwargs == {
+        "ml_model": "gpt-4o-mini",
+        "input_values": [
+            PromptRequestStringInput(key="question", type="STRING", value="What's the weather like in San Francisco?"),
+            PromptRequestChatHistoryInput(
+                key="chat_history",
+                type="CHAT_HISTORY",
+                value=[
+                    ChatMessage(role="USER", text="Hello, how are you?"),
+                    ChatMessage(role="ASSISTANT", text="I'm good, thank you!"),
+                ],
+            ),
+        ],
+        "input_variables": [
+            VellumVariable(
+                id=str(first_call_input_id),
+                key="question",
+                type="STRING",
+                required=None,
+                default=None,
+                extensions=None,
+            ),
+            VellumVariable(
+                id=str(first_call_input_id_2),
+                key="chat_history",
+                type="CHAT_HISTORY",
+                required=None,
+                default=None,
+                extensions=None,
+            ),
+        ],
+        "parameters": DEFAULT_PROMPT_PARAMETERS,
```

Comment:
[nit] feel free to only have assertions for:
- `input_values`
- `input_variables`
- `blocks`